client/client.go

```diff
@@ -46,57 +47,90 @@ func NewKavaClient(cdc *amino.Codec, mnemonic string, coinID uint32, rpcAddr str
 
 // Broadcast sends a message to the Kava blockchain as a transaction.
 // This pays no transaction fees.
-func (kc *KavaClient) Broadcast(m sdk.Msg, syncType SyncType) (*ctypes.ResultBroadcastTx, error) {
-	fee := authtypes.NewStdFee(250000, nil)
-	return kc.BroadcastWithFee(m, fee, syncType)
+func (kc *KavaClient) Broadcast(clientCtx client.Context, m sdk.Msg, syncType SyncType) (res *sdk.TxResponse, err error) {
+	fee := legacytx.NewStdFee(250000, nil)
+	return kc.BroadcastWithFee(clientCtx, m, fee, syncType)
 }
 
 // BroadcastWithFee sends a message to the Kava blockchain as a transaction, paying the specified transaction fee.
-func (kc *KavaClient) BroadcastWithFee(m sdk.Msg, fee authtypes.StdFee, syncType SyncType) (*ctypes.ResultBroadcastTx, error) {
+func (kc *KavaClient) BroadcastWithFee(clientCtx client.Context, m sdk.Msg, fee legacytx.StdFee, syncType SyncType) (res *sdk.TxResponse, err error) {
 	signBz, err := kc.sign(m, fee)
 	if err != nil {
 		return nil, err
 	}
+
+	// build a legacy transaction
+	sigs := []legacytx.StdSignature{legacytx.NewStdSignature(kc.Keybase.GetKeyRing().GetPubKey(), signBz)}
+	stdTx := legacytx.NewStdTx([]sdk.Msg{m}, fee, sigs, "")
+	stdTx.TimeoutHeight = 100000
+
+	var mode string
 	switch syncType {
 	case Async:
-		return kc.BroadcastTxAsync(signBz)
+		mode = "async"
 	case Sync:
-		return kc.BroadcastTxSync(signBz)
+		mode = "sync"
 	case Commit:
-		commitRes, err := kc.BroadcastTxCommit(signBz)
+		mode = "commit"
```

Comment:
Commit mode should be replaced with Block:

https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/client/flags/flags.go#L24-L32

https://github.com/cosmos/cosmos-sdk/blob/v0.44.5/client/broadcast.go#L23-L39

Currently returns `unsupported return type commit; supported types: sync, async, block` if Commit mode is used